rules/prefer-class-fields.js

```diff
@@ -0,0 +1,116 @@
+'use strict';
+const getIndentString = require('./utils/get-indent-string.js');
+
+const MESSAGE_ID = 'prefer-class-fields/error';
+const messages = {
+	[MESSAGE_ID]:
+		'Prefer class field declaration over `this` assignment in constructor for static values.',
+};
+
+/**
+ * @param {import('eslint').Rule.Node} node
+ * @returns {node is import('estree').ExpressionStatement & {expression: import('estree').AssignmentExpression & {left: import('estree').MemberExpression & {object: import('estree').ThisExpression}}}}
+ */
+const isThisAssignmentExpression = node => {
+	if (
+		node.type !== 'ExpressionStatement'
+		|| node.expression.type !== 'AssignmentExpression'
+	) {
+		return false;
+	}
+
+	const lhs = node.expression.left;
+
+	if (!lhs.object || lhs.object.type !== 'ThisExpression') {
```

Comment:
The left property didn't checked

```js
			class Foo {
				constructor() {
					this[foo.bar] = 'foo';
				}
			}
```

Reply:
I think in this case we should ignore any kind of reporting - `foo.bar` might be dynamic and checking if it is or not is a rabbit hole with potentially lots of nested jumps.

Reply:
Added a valid test case to cover the dynamic case: 633829c7daf29bd071a2ed371bbee6de5f28cd9a

Reply:
And here: 0e6afb3a27ad2de8ccb5dc8cc7dcb2e89261863c added support for simple strings & template strings.

Reply:
I didn't mean to ask support for strings, please test

```js
			class Foo {
				constructor() {
					this[foo] = 'foo';
				}
			}
```

Reply:
Don't worry, I didn't understood it well then :)
Done: f9caf1b8fdf8f85b87c2036023b6bf2e7972e476